tests/StockTest.php

```diff
@@ -22,4 +22,18 @@ public function testAdd()
         $stock_array = $property->getValue($this->stock);
         $this->assertEquals($element, $stock_array[0]);
     }
+    public function testTake()
+    {
+        $element = 'silnik';
+        $this->stock->add($element);
+        $taken_element = $this->stock->take();
+
+        $class = new ReflectionClass($this->stock);
+
+        $property = $class->getProperty('stock');
+        $property->setAccessible(true);
+        $stock_array = $property->getValue($this->stock);
+        $this->assertEquals(0,count($stock_array));
+        $this->assertEquals($taken_element, 'silnik');
```

Comment:
duplikacja kodu.
Zauważ, że 'silnik' masz przypisany do zmiennej $element.
Zamiast ponownie jawnie pisać string to możesz użyć zmienej, a nawet powinieneś.

`$this->assertEquals($taken_element, $element);`